### PR TITLE
Factor out release

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
Nightly releases are not working reliably.

# Changes
- Factor out the release code into an action
- Call the action from nightly
- Increase the frequency of the nightly job to hourly

# Tests
Ultimately:  Delete the release for today, wait an hour, check.

I have triggered the build manually but there is no guarantee that manual triggers behave in the same way.

### CODE REVIEW

File: .github/workflows/nightly.yaml [hunk 1]
```diff
@@ -28,7 +29,29 @@ jobs:
             tag_name="$(date +nightly-%Y-%m-%d)"
             git tag "${tag_name}"
             git push origin "refs/tags/${tag_name}"
```
Comment (bitdivine) [inline .github/workflows/nightly.yaml:31]: It may be that teh tag already exists, just not on this commit.

**Review by lmuntaner — APPROVED** [review]
A couple of questions, but LGTM, thanks!

File: .github/actions/release_nns_dapp/action.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+name: 'Release the nns-dapp'
+description: |
+  Creates or updates a release for every tag pointing at HEAD.
```
[the remark this answers is not in the document]
Comment (bitdivine) [reply .github/actions/release_nns_dapp/action.yaml:3]: It makes a release for all tags pointing at head, no exceptions.  It is worth mentioning that the nightly- and proposal- releases are also made public.  I'll add that.  Done.

File: .github/workflows/nightly.yaml [hunk 3]
```diff
@@ -27,8 +28,32 @@ jobs:
...
             tag_name="$(date +nightly-%Y-%m-%d)"
             git tag "${tag_name}"
-            git push origin "refs/tags/${tag_name}"
+            if git push origin "refs/tags/${tag_name}"
+            then echo "need_release=true" >> "$GITHUB_OUTPUT"
+            else echo "Skipping, as the tag already exists on another commit."
+            fi
           fi
+      - name: Set up docker buildx
+        if: steps.tag_main.outputs.need_release == 'true'
```
[the remark this answers is not in the document]
Comment (bitdivine) [reply .github/workflows/nightly.yaml:37]: From line 32:
> `then echo "need_release=true" >> "$GITHUB_OUTPUT"`

File: .github/actions/release_nns_dapp/action.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (lmuntaner) [reply .github/actions/release_nns_dapp/action.yaml:3]: Ah, understood. That's the difference, public or not.

Thanks!

File: .github/workflows/nightly.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (lmuntaner) [reply .github/workflows/nightly.yaml:37]: ok, thanks for the explanation!

[the remark this answers is not in the document]
Comment (bitdivine) [reply .github/workflows/nightly.yaml:37]: Ah, the example is at job level but the task level is similar.  Let me see if I can find an example in the documentation.

[the remark this answers is not in the document]
Comment (lmuntaner) [reply .github/workflows/nightly.yaml:37]: No worries, I understand the idea. Thanks!

**Pull request merged**